presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -528,4 +565,44 @@ public Boolean process(Node node, @Nullable Void context)
             return super.process(node, context);
         }
     }
+
+    /**
+     * Verifies that no query node output columns are referenced from ORDER BY expressions.
+     * Only FROM columns are allowed within ORDER BY expressions.
+     */
+    private class VerifyOrderByAggregationVisitor
+            extends DefaultTraversalVisitor<Void, Void>
+    {
+        @Override
+        protected Void visitIdentifier(Identifier node, Void context)
+        {
+            verifyField(node, QualifiedName.of(node.getName()));
+            return null;
+        }
+
+        @Override
+        protected Void visitDereferenceExpression(DereferenceExpression node, Void context)
+        {
+            if (analysis.getColumnReferences().contains(node)) {
+                verifyField(node, DereferenceExpression.getQualifiedName(node));
+            }
+
+            return null;
```

Comment:
You need to call `process(..)` here. If `a` or `t.a` would be prohibited, you need to report also if `a.field` or `t.a.field` is used. Or `foo(a).field`.